lib/cocoapods/installer/pod_source_installer.rb

```diff
@@ -194,13 +190,10 @@ def released?
         !local? && !predownloaded? && sandbox.specification(root_spec.name) != root_spec
       end
 
-      def each_source_file(file_accessors, &blk)
-        file_accessors.each do |file_accessor|
-          file_accessor.source_files.each do |source_file|
-            next unless source_file.exist?
-            blk[source_file]
-          end
-        end
+      # @return [Array<Pathname>] The paths of the source files
+      #
+      def source_files(file_accessors)
+        @source_files ||= file_accessors.flat_map(&:source_files)
```

Comment:
this probably needs to key off of the file accessors, as we should always return the correct result even if given 2 different arrays of file accessors

Reply:
good point - I think I'll hold off on this part for now then and revisit it again in another PR